atoma-proxy/src/server/streamer.rs

```diff
@@ -340,6 +360,13 @@ impl Stream for Streamer {
 
                 let chunk = match serde_json::from_str::<Value>(chunk_str) {
                     Ok(chunk) => {
+                        if let Some(timer) = self.inter_stream_token_latency_timer.take() {
```

Comment:
I believe you need to set a new timer instant for the next token

Reply:
Good catch! added 3c1659a